spec/test_application/app/controllers/test_application/base_controller.rb

```diff
@@ -59,6 +59,10 @@ def explode_with_unhandled_exception
 
     private
 
+    def jsonapi_options(object)
+      { jsonapi: object }.merge(content_type: 'application/vnd.api+json')
+    end
+
```

Comment:
You don't need this method and also you should have written 
```
{ jsonapi: object, content_type: 'application/vnd.api+json' }
```
instead of using `merge`.

Reply:
I tried out these changes myself and found a bug in `Shark::BaseController`. It should be 

```diff
     ActionController::Renderers.add :jsonapi do |object, options|
-       response.set_header('content-type', 'application/vnd.api+json')     
+      response.content_type = 'application/vnd.api+json; charset=utf-8'
```

Now every call to `render jsonapi: object` also sets the correct content type and nobody needs the override `render` anymore.